### ISSUE TITLE
Getting state and sending messages outside of message handler

### ISSUE THREAD

**Issue by 13g10n** [aiogram/aiogram#1045]
### aiogram version

3.x

### Problem

Hi,

I'm getting in troubles while trying to figure out how to send messages or get state outside of mesage handler function.

In `2.x` branch we had `Dispatcher.get_current()`, `Chat.get_current()`, `Dispatcher.get_current().current_state()` that works perfectly in my cases, but it seems we don't have an alternative in `3.x`.

### Possible solution

The only solution I can see now if to pass `Message` and `FSMContext` objects to my custom service, but it looks a bit ugly when it comes to complex logic with a lot of calls and/or custom plugins. 

### Code example

```python3
@router.message(Command(commands=['start']))
async def command_start(message: Message, state: FSMContext) -> None:
    # Basically, I want to clean up logic and remove this parameters in `foo` service method
    await custom_service.foo(message, state)
```

### Additional information

BTW, thanks for the project - great work!

**Comment by 13g10n** [comment]
Thanks for the `resolve_context` method, missed it during my investigations.

BTW, perfect solution for my plugin was exactly to use **Dependency injection** via **custom middleware**.
